### DISCUSSION TITLE
Using with react-native-ble-plx

### DISCUSSION THREAD

**Discussion by jmlarbaig** [tijmenvangulik/ErgometerJS#24] [Q&A]
Hello Tijmen

I use react native ble plx library until now and I would like to use your library, 

Do you think that it's will be possible ?

**Comment by tijmenvangulik** [comment]
react-native-ble-plx has an javascript/ Typescript interface. ErgometerJs is made for talking to all kinds of libraries. It should not be difficult to change it. You have to implement your own driver (implement the interface IDriver).
 Look at the source of The DriverBleCentral or DriverSimleBLE or DriverWebBlueTooth to see how to do this. The library ( PerformanceMonitorBle) will not directly recognize your new driver but you can set the monitor using the driver property:

public set driver(value:ble.IDriver) {
            
            this._driver = value;
            
        }

**Reply by jmlarbaig** [reply]
Ok, I see what you mean. After that, If I understand well, I just need to replace bleat variable by the new one ?

**Comment by tijmenvangulik** [comment]
I think you are talking about de bleat include in the demo. Yes you can remove that one and also the noble include. Just use your own library. If you get the driver working , I am interested in your source code, I could publish it as part of ErgometerJs. (And as advantage for you I will check your code)

var noble = require('noble');
var bleat = require('./libs/bleat/index').classic;

**Reply by jmlarbaig** [reply]
Great, so I will try to create new driver so. 

I will be back if something work,

Thanks for your fast answer,

**Reply by jmlarbaig** [reply]
I'm sorry, It's will be the first time when I need to do this kind of operation. 

I have found somes files and driver (in ergometer.js and in the directory of api with different files of BLE). Could you tell me which files I can see the sources of Driver already create please,

**Comment by tijmenvangulik** [comment]
It is best to check the typescript files which are compiled into javascript. These contain the source code:

You will need  to create a  ble driver which are located here: (The driver.ts contains the interface IDriver and the other files are different implementations )

ErgometerJS/api/typescript/ergometer/ble at master · tijmenvangulik/ErgometerJS (github.com) <https://github.com/tijmenvangulik/ErgometerJS/tree/master/api/typescript/ergometer/ble>

There are also usb drivers but you can ignore these. You want ble.

ErgometerJS/api/typescript/ergometer/usb at master · tijmenvangulik/ErgometerJS (github.com) <https://github.com/tijmenvangulik/ErgometerJS/tree/master/api/typescript/ergometer/usb>

**Reply by jmlarbaig** [reply]
Sorry ! I thought it's was ble-plx library what I use but this is this one : https://github.com/innoveit/react-native-ble-manager

It's a portion of BleCentral you mean in your different driver. May be it's will be work so.

**Comment by tijmenvangulik** [comment]
Then it should not be difficult to convert the ble central driver  It will not work directly you need some changes to connect the driver to the react native ble manager. Often the best approach is to first get react native ble manger running without ergometer space and connect to the ergometer and once that is running, make the changes to let driver communicate with the react native ble manager. This way you are more sure where the problems are when you can not connect.

react-native-ble-manager/BleManager.js at master · innoveit/react-native-ble-manager (github.com) <https://github.com/innoveit/react-native-ble-manager/blob/master/BleManager.js>

Tijmen

**Reply by jmlarbaig** [reply]
I have tested this module and with your extension ergometer.js. On simulation it worked fine. 

I will test it with PM2 when I will have one,
